core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -106,3 +106,45 @@ message TaskDataWrapper {
   int64 stage_id = 40;
   int32 stage_attempt_id = 41;
 }
+
+message ExecutorResourceRequest {
+  string resource_name = 1;
+  int64 amount = 2;
+  string discoveryScript = 3;
+  string vendor = 4;
+}
+
+message TaskResourceRequest {
+  string resource_name = 1;
+  double amount = 2;
+}
+
+message ResourceProfileInfo {
+  int32 id = 1;
+  map<string, ExecutorResourceRequest> executor_resources = 2;
+  map<string, TaskResourceRequest> task_resources = 3;
+}
+
+message RuntimeInfo {
+  string java_version = 1;
+  string java_home = 2;
+  string scala_version = 3;
+}
+
+message ApplicationEnvironmentInfo {
+  message PairSS {
```

Comment:
Shall we name it `PairStrings` and move it out of `ApplicationEnvironmentInfo`? In the future this might be reused.